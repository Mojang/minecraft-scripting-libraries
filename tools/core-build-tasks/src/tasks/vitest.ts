import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
