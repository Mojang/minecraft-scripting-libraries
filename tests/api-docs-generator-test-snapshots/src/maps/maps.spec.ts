// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Maps', () => {
    it('Properly generates documentation for maps', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['ts', 'msdocs'],
        });
    });
});
