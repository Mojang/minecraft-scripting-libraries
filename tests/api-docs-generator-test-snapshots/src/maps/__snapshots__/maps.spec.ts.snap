// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Maps > Properly generates documentation for maps > # Of Files Generated 1`] = `4`;

exports[`Maps > Properly generates documentation for maps > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: maps-module
    items:
    - name: maps-module
      href: maps-module/maps-module.md
    - name: ExampleClass
      href: maps-module/ExampleClass.md
"
`;

exports[`Maps > Properly generates documentation for maps > msdocs/scriptapi/maps-module/ExampleClass.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: maps-module.ExampleClass Class
description: Contents of the maps-module.ExampleClass class.
---
# ExampleClass Class

## Properties

### **mapProperty**
\`mapProperty: Record<string, number>;\`

Type: Record<*string*, *number*>

Notes:
  - This property can't be used in read-only mode.

## Methods
- [exampleMethodReturnsMap](#examplemethodreturnsmap)
- [exampleMethodTakesMapArg](#examplemethodtakesmaparg)

### **exampleMethodReturnsMap**
\`
exampleMethodReturnsMap(): Record<string, number>
\`

**Returns** Record<*string*, *number*>
  
Notes:
- This function can't be called in read-only mode.

### **exampleMethodTakesMapArg**
\`
exampleMethodTakesMapArg(arg: Record<string, number>): void
\`

#### **Parameters**
- **arg**: Record<*string*, *number*>
  
Notes:
- This function can't be called in read-only mode.
"
`;

exports[`Maps > Properly generates documentation for maps > msdocs/scriptapi/maps-module/maps-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: maps-module Module
description: Contents of the maps-module module
---
# \`maps-module\` Module

## Manifest Details
\`\`\`json
{
    "module_name": "maps-module",
    "version": "1.0.0"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0\`

## Classes
- [ExampleClass](ExampleClass.md)
"
`;

exports[`Maps > Properly generates documentation for maps > typescript/maps-module@1.0.0.d.ts 1`] = `
"// Type definitions for Minecraft Bedrock Edition script APIs
// Project: https://docs.microsoft.com/minecraft/creator/
// Definitions by: Jake Shirley <https://github.com/JakeShirley>
//                 Mike Ammerlaan <https://github.com/mammerla>

/* *****************************************************************************
   Copyright (c) Microsoft Corporation.
   ***************************************************************************** */
/**
 * @packageDocumentation
 *
 * Manifest Details
 * \`\`\`json
 * {
 *   "module_name": "maps-module",
 *   "version": "1.0.0"
 * }
 * \`\`\`
 *
 */
export class ExampleClass {
    private constructor();
    /**
     * @remarks
     * This property can't be used in read-only mode.
     *
     */
    mapProperty: Record<string, number>;
    /**
     * @remarks
     * This function can't be called in read-only mode.
     *
     */
    exampleMethodReturnsMap(): Record<string, number>;
    /**
     * @remarks
     * This function can't be called in read-only mode.
     *
     */
    exampleMethodTakesMapArg(arg: Record<string, number>): void;
}
"
`;
